packages/react-router-dom/index.tsx

```diff
@@ -378,22 +413,26 @@ export function useSearchParams(defaultInit) {
  *     sort: ['name', 'price']
  *   });
  */
-export function createSearchParams(init = '') {
-  if (
-    typeof init !== 'string' &&
-    !Array.isArray(init) &&
-    !(init instanceof URLSearchParams)
-  ) {
-    init = Object.keys(init).reduce((memo, key) => {
-      let value = init[key];
-      if (Array.isArray(value)) {
-        value.forEach(v => memo.push([key, v]));
-      } else {
-        memo.push([key, value]);
-      }
-      return memo;
-    }, []);
-  }
-
-  return new URLSearchParams(init);
+export function createSearchParams(
```

Comment:
I may eventually go back to using multiple files in the codebase, but I prefer to group the code by bundle for now. It's just a little easier for me to work with it this way because I can more easily see which pieces are going into which bundle (i.e. the `index` vs. `server` bundles).